source/common/common/assert.h

```diff
@@ -91,20 +91,36 @@ void invokeDebugAssertionFailureRecordAction_ForAssertMacroUseOnly();
 // correctly when compiled with MSVC
 #define EXPAND(X) X
 
+// This non-implementation ensures that its argument is a valid expression that can be statically
+// casted to a bool, but the expression is never evaluated and will be compiled away.
+#define _NULL_ASSERT_IMPL(X, ...)                                                                  \
+  do {                                                                                             \
+    constexpr bool __assert_dummy_variable = false && static_cast<bool>(X);                        \
+    (void)__assert_dummy_variable;                                                                 \
+  } while (false)
+
+#if defined(ENVOY_DEBUG_KNOWN_ISSUES)
+/**
+ * Assert wrapper for an as-yet unidentified issue. Even with ASSERTS compiled in, it will be
+ * excluded, unless ENVOY_DEBUG_KNOWN_ISSUES is defined. It represents a condition that
+ * should always pass but that sometimes fails for an unknown reason. The macro allows it to
+ * be temporarily compiled out while the failure is triaged and investigated.
+ */
+#define KNOWN_ISSUE_ASSERT(X, DETAILS) _ASSERT_IMPL(X, #X, abort() DETAILS)
```

Comment:
Per above I think we should invert the logic. Still, I realize now that you are trying to disable a normal ASSERT vs. a RELEASE_ASSERT. Why are you shipping code with normal ASSERTs enabled? Are you using an optimized built with regular asserts on? Why not just turn them off?

Reply:
It has a lot to do with the early stage Envoy Mobile is in, and the fact that it's under active development. We're calling into Envoy via new mechanisms and in new contexts. Sometimes expectations fail to hold. Often it's because we've erred, and sometimes it's because we've uncovered a new edge case. Either way, at this point, we'd like to maintain that visibility by keeping (most) assertions compiled in. For now.